lib/src/types/range_types.dart

```diff
@@ -0,0 +1,258 @@
+import 'package:meta/meta.dart';
+
+/// Describes PostgreSQL range bound state
+///
+/// https://www.postgresql.org/docs/current/rangetypes.html#RANGETYPES-INCLUSIVITY
+enum Bound {
+  /// Equivalent to '(' or ')'
+  exclusive,
+
+  /// Equivalent to '[' or ']'
+  inclusive,
+}
+
+/// Describes the bounds of PostgreSQL range types.
+///
+/// https://www.postgresql.org/docs/current/rangetypes.html#RANGETYPES-INCLUSIVITY
+@immutable
+class Bounds {
+  late final Bound lower;
+  late final Bound upper;
+
+  Bounds(this.lower, this.upper);
+
+  Bounds.fromFlag(int flag) {
+    switch (flag) {
+      case 0 || 1:
```

Comment:
This switch-case structure is a bit obscure, what is the source of it? Can we do bit operations instead?

Reply:
I added a comment explaining the constructor and made switch statement shorter. I'm unfamiliar with bit operations so I would prefer to leave it as is for now.